File: frontend/lib/models/travel_metadata.dart
class TravelPurpose {
  const TravelPurpose({
    required this.id,
    required this.name,
    this.description,
  });

  factory TravelPurpose.fromJson(Map<String, dynamic> json) {
    return TravelPurpose(
      id: json['_id'] as String,
      name: json['name'] as String? ?? '',
      description: json['description'] as String?,
    );
  }

  final String id;
  final String name;
  final String? description;
}

class AccommodationType {
  const AccommodationType({
    required this.id,
    required this.name,
    this.description,
  });

  factory AccommodationType.fromJson(Map<String, dynamic> json) {
    return AccommodationType(
      id: json['_id'] as String,
      name: json['name'] as String? ?? '',
      description: json['description'] as String?,
    );
  }

  final String id;
  final String name;
  final String? description;
}

class TravelAmenity {
  const TravelAmenity({
    required this.id,
    required this.name,
  });

  factory TravelAmenity.fromJson(Map<String, dynamic> json) {
    return TravelAmenity(
      id: json['_id'] as String,
      name: json['name'] as String? ?? '',
    );
  }

  final String id;
  final String name;
}

File: backend/src/models/TravelMetadata.js
const mongoose = require('mongoose');

const travelPurposeSchema = new mongoose.Schema(
  {
    name: { type: String, required: true, unique: true },
    description: { type: String, default: '' },
  },
  { timestamps: true }
);

const accommodationTypeSchema = new mongoose.Schema(
  {
    name: { type: String, required: true, unique: true },
    description: { type: String, default: '' },
  },
  { timestamps: true }
);

const amenitySchema = new mongoose.Schema(
  {
    name: { type: String, required: true, unique: true },
  },
  { timestamps: true }
);

const TravelPurpose = mongoose.model('TravelPurpose', travelPurposeSchema);
const AccommodationType = mongoose.model('AccommodationType', accommodationTypeSchema);
const TravelAmenity = mongoose.model('TravelAmenity', amenitySchema);

module.exports = {
  TravelPurpose,
  AccommodationType,
  TravelAmenity,
};

